ProgramHierarchy.java

```diff
@@ -1,18 +1,44 @@
-/*
- * File: ProgramHierarchy.java
- * Name: 
- * Section Leader: 
- * ---------------------------
- * This file is the starter file for the ProgramHierarchy problem.
- */
-
-import acm.graphics.*;
-import acm.program.*;
-import java.awt.*;
-
-public class ProgramHierarchy extends GraphicsProgram {	
-	public void run() {
-		/* You fill this in. */
-	}
-}
-
+/*
+ * File: ProgramHierarchy.java
+ * Name: 
+ * Section Leader: 
+ * ---------------------------
+ * This file is the starter file for the ProgramHierarchy problem.
+ */
+
+import acm.graphics.*;

```

Comment:
Generally you should avoid the use of constant numbers in your program. You have values like (30, 130, 175, 75...etc) throughout your code and this is not a good idea because:

1. Someone reading your code will be confused how you came up with them
2. If requirements change your program will be really hard to update. For example imagine for this exercise the requirements changed and we decide that the value for HEIGHT (50) and WIDTH (150) should be changed to HEIGHT = 60 and WIDTH = 180

If this happens your program will fail to draw the right diagram. To make it work you will have to go back and start changing all the numbers throughout your program. It should be possible to come up with a solution that works without you doing further modification. The ability of a program to be able to adapt to changes like that is called scalability. So in this case your solution is not scalable enough.